model/labels/labels.go

```diff
@@ -336,16 +336,29 @@ func (ls Labels) Validate(f func(l Label) error) error {
 	return nil
 }
 
-// DropMetricName returns Labels with "__name__" removed.
+// DropMetricName returns Labels with the "__name__" removed.
+// Deprecated: Use DropSpecial instead.
 func (ls Labels) DropMetricName() Labels {
+	return ls.DropSpecial(func(n string) bool { return n == MetricName })
+}
+
+// DropSpecial returns Labels without the chosen (via shouldDropFn) special (starting with underscore) labels.
+func (ls Labels) DropSpecial(shouldDropFn func(name string) bool) Labels {
```

Comment:
```suggestion
func (ls Labels) DropReservedLabels(shouldDropFn func(name string) bool) Labels {
```

Not sure if I'm the only one, but `DropSpecial` sounds weird 😅